tests/test_implements_satisfy_protocols.py

```diff
@@ -152,27 +155,9 @@ class RuntimeCheckableDimension(DimensionProtocol, Protocol):
     pass
 
 
-def test_dimension_protocol() -> None:  # noqa: D
-    time_dim = PydanticDimension(
-        name="test_time_dim",
-        type=DimensionType.TIME,
-        label="Test Time Dim",
-        type_params=PydanticDimensionTypeParams(
-            time_granularity=TimeGranularity.DAY,
-            validity_params=PydanticDimensionValidityParams(),
-        ),
-    )
-    assert isinstance(time_dim, RuntimeCheckableDimension)
-
-    # Skipping this assertion because are implementation of the function `time_dimension_reference` raises an
-    # exception if DimensionType != TIME. The isinstance check seems to actually run the function thus
-    # raising an exception during the assertion.
-    # of
-    # categorical_dim = PydanticDimension(
-    #     name="test_categorical_dim",
-    #     type=DimensionType.CATEGORICAL,
-    # )
-    # assert isinstance(categorical_dim, RuntimeCheckableDimension)
+@given(builds(PydanticDimension))
+def test_dimension_protocol(dimesnion: PydanticDimension) -> None:  # noqa: D
+    assert isinstance(dimesnion, RuntimeCheckableDimension)
```

Comment:
How is this not subject to the same problem with categorical dimension types as the old test?

Reply:
Thats a good question. So I ran the test a few different ways after seeing your comment in this review.

1.
```python
import pdb
@given(builds(PydanticDimension))
def test_dimension_protocol(dimension: PydanticDimension) -> None:  # noqa: D
    if dimension.type == DimensionType.CATEGORICAL:
        pdb.set_trace()
    assert isinstance(dimension, RuntimeCheckableDimension)
```

2.
```python
import pdb
@given(builds(PydanticDimension))
def test_dimension_protocol(dimension: PydanticDimension) -> None:  # noqa: D
    if dimension.type == DimensionType.TIME:
        pdb.set_trace()
    assert isinstance(dimension, RuntimeCheckableDimension)
```

3.
```python
import pdb
@given(builds(PydanticDimension))
def test_dimension_protocol(dimension: PydanticDimension) -> None:  # noqa: D
    if dimension.type_params is not None:
        pdb.set_trace()
    assert isinstance(dimension, RuntimeCheckableDimension)
```

Variation (1) and (2) hit the break point when the test was ran, but (3) never was. Thus it appears the type_params aren't getting automatically generated by hypothesis. I'm gonna do some more investigating and see if this is happening to _all nested objects_. If so I'll have to build better strategies. I haven't found a setting in hypothesis to handle this better by default. It looks like we might have be able to [register strategies](https://hypothesis.readthedocs.io/en/latest/data.html#hypothesis.strategies.register_type_strategy) for custom types, which is maybe what we have to do?

Reply:
Digging in further, this might be a pydantic class defaulting issue https://github.com/HypothesisWorks/hypothesis/issues/3218. Which makes sense why I was seeing it on some things, but not on others.

Reply:
So I improved our strategies, but that was a separate fix. Why the tests no longer have the issue with categorical dimensions is we removed the assertion causing the issue in this commit https://github.com/dbt-labs/dbt-semantic-interfaces/commit/83cc44d2d896892e756cfd6a17f5bbe0456821a4